openssl/src/x509/mod.rs

```diff
@@ -129,32 +129,34 @@ impl X509StoreContext {
 /// let cert_pem = cert.to_pem().unwrap();
 /// let pkey_pem = pkey.private_key_to_pem().unwrap();
 /// ```
-pub struct X509Generator {
+pub struct X509Generator<'ctx> {
```

Comment:
That's OK. If I understand well, the lifetime specifier <'ctx> breaks the
some implementation. All previous tests do work without any modification.
You changed the interface and removed the generate() function, so
everything is done through the sign function. Maybe I can use the sign
function of the CA to sign my SSL certificate and refactor the code like
that.
I'll check that.

2016-08-14 18:31 GMT+02:00 Steven Fackler <notifications@github.com>:

> In openssl/src/x509/mod.rs
> <https://github.com/sfackler/rust-openssl/pull/411#discussion_r74704755>:
>
> > @@ -129,32 +129,34 @@ impl X509StoreContext {
> >  /// let cert_pem = cert.to_pem().unwrap();
> >  /// let pkey_pem = pkey.private_key_to_pem().unwrap();
> >  /// ```
> > -pub struct X509Generator {
> > +pub struct X509Generator<'ctx> {
>
> Adding this type parameter is unfortunately a breaking change. Maybe we
> should have set_ca take the cert and pkey by value and fix it when we cut
> the next breaking release?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sfackler/rust-openssl/pull/411/files/8d9273ba3de769dc580363447d0dd4fdf14329e7#r74704755>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABEe8_NlppjoTX3LXchn_gVA1vemeMC7ks5qf0LNgaJpZM4I20vR>
> .
>